gtsam/hybrid/tests/testHybridFactorGraph.cpp

```diff
@@ -335,10 +337,68 @@ TEST(DCGaussianElimination, Eliminate_fully) {
   // EXPECT_LONGS_EQUAL(1, result.second->size());
 }
 
+/* ****************************************************************************/
+/// Test the toDecisionTreeFactor method
+TEST(HybridFactorGraph, ToDecisionTreeFactor) {
+  size_t K = 3;
+
+  // Provide tight sigma values so that the errors are visibly different.
+  double between_sigma = 5e-8, prior_sigma = 1e-7;
+
+  Switching self(K, between_sigma, prior_sigma);
+
+  // Clear out discrete factors since sum() cannot hanldle those
+  HybridFactorGraph linearizedFactorGraph(
+      NonlinearFactorGraph(), DiscreteFactorGraph(),
+      self.linearizedFactorGraph.dcGraph(),
+      self.linearizedFactorGraph.gaussianGraph());
+
+  auto decisionTreeFactor = linearizedFactorGraph.toDecisionTreeFactor();
+
+  auto allAssignments = cartesianProduct(linearizedFactorGraph.discreteKeys());
+
+  // Get the error of the discrete assignment m1=0, m2=1.
+  double actual = (*decisionTreeFactor)(allAssignments[1]);
+
+  /********************************************/
+  // Create equivalent factor graph for m1=0, m2=1
+  GaussianFactorGraph graph;
+
+  // Create the linearization point.
+  Values linearizationPoint;
```

Comment:
Cool test! Note that `self.linearizationPoint` exists so no need for this. Can just fold into next PR.